### ISSUE TITLE
Format SQL Statement fails

### ISSUE THREAD

**Issue by ehamberg** [jakob/Postico#851]
## What did you do?

I hit ⌃F to format a statement.

## What did you expect to happen?

- I expected my statement to be formatted.

## What actually happened?

- The app beeped at me, and nothing was formatted.

## Suspected root cause

I'm using [home-manager]() to manager my system, so `~/.pg_format` is symlinked to `/nix/store/844xrg862v21xh93dx0qxxvj71lhbmr3-home-manager-files/.pg_format`.

If I remove the symlink and copy the file to `~/.pg_format`, things work again. It even works if I rename the file, keep it in my homedir and symlink `~/.pg_format` to _that_ file.

This leads me to suspect that this is caused by sandboxing and that macOS sees that Postico is trying to access a file on a different volume. It's still a bit curious, though, because it fails even if I grant Postico full disk access.

## What software versions are you using?

Postico version:  
Version 2.0.4 (9636)

macOS version:  
13.5.2 (22G91)

PostgreSQL version:  
PostgreSQL 14.9 (Debian 14.9-1.pgdg120+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit

## Postscript

Feel free to close this unless there is something interesting here. This is probably not a very common setup. :)

**Comment by jakob** [comment]
Postico should probably show the stderr output of pg_format instead of beeping. However, you can see the error message from pg_format if you start Postico from the command line like this:

```sh
$ /Applications/Postico\ 2/Contents/MacOS/Postico
```

If it's the sandbox, you should also see a message in Console.app with the reason why it was denied. Unfortunately filtering in Console.app is a bit difficult, but searching for "sandbox" should show the relevant log lines.

**Comment by ehamberg** [comment]
> Postico should probably show the stderr output of pg_format instead of beeping. 

Yep. That would have helped here – or any way told me that an error occurred. When my Mac is muted (which it often is), nothing happens. :)

> However, you can see the error message from pg_format if you start Postico from the command line like this:
> 
> ```shell
> $ /Applications/Postico\ 2/Contents/MacOS/Postico
> ```

Ah! That works. Here's the output:

```
Uncaught exception: Can't open '/Users/ehamberg/.pg_format' for reading: 'Operation not permitted' at /Applications/Postico 2.app/Contents/Resources/lib/pgFormatter/CLI.pm line 365
 at /Applications/Postico 2.app/Contents/Resources/pg_format line 19.
	main::__ANON__(autodie::exception=HASH(0x14689e9c8)) called at (eval 7) line 154
	pgFormatter::CLI::__ANON__(GLOB(0x1468342a8), "<", "/Users/ehamberg/.pg_format") called at /Applications/Postico 2.app/Contents/Resources/lib/pgFormatter/CLI.pm line 365
	pgFormatter::CLI::get_command_line_args(pgFormatter::CLI=HASH(0x146811378)) called at /Applications/Postico 2.app/Contents/Resources/lib/pgFormatter/CLI.pm line 62
	pgFormatter::CLI::run(pgFormatter::CLI=HASH(0x146811378)) called at /Applications/Postico 2.app/Contents/Resources/pg_format line 49
```

> If it's the sandbox, you should also see a message in Console.app with the reason why it was denied. Unfortunately filtering in Console.app is a bit difficult, but searching for "sandbox" should show the relevant log lines.

Ah. I tried searching for Postico without success. Here's the relevant line from Console.app:

```
error	11:32:19.608275+0200	kernel	Sandbox: perl5.30(96447) deny(1) file-read-data /nix/store/ky7b87nrn5hgmv64s9yi2nmrj5241dpb-hm_pg_format
```

**Comment by jakob** [comment]
Thanks for the follow-up.

I've been working on a feature to fix this problem. I just merged it into the development branch, which you can download here:

https://releases.eggerapps.at/postico2/changelog?update_channel=1

Build 9658 adds a security preference pane that allows you to give Postico access to certain files. The feature was meant for people who use symbolic links / aliases to organise their query files, but it could also help with the .pg_format symlink.

You can use the new security pref pane in Postico to give Postico permission to eg. `/nix/store`, and then it should work. You can then remove the full disk access permission.

**Comment by ehamberg** [comment]
Woo! That worked perfectly. Thanks!

(Not sure if I should close this now, or if you close issues once they're fixed in the stable channel.)
